pulse2percept/retina.py

```diff
@@ -27,158 +33,246 @@ def __init__(self, xlo=-1000, xhi=1000, ylo=-1000, yhi=1000, eye='RE',
+            self.axon_distances = utils.parfor(axon_dist_from_soma, self.axons,
+                                               func_args=[xg, yg],
+                                               func_kwargs=func_kwargs,
+                                               engine=engine, n_jobs=n_jobs,
+                                               scheduler=scheduler)
+            grid_dict['axon_distances'] = self.axon_distances
 
             # Save the variables, together with metadata about the grid:
             if save_data:
-                np.savez(filename,
-                         eye=[eye],
-                         axon_id=ax_id,
-                         axon_weight=ax_wt,
-                         jan_x=jan_x,
-                         jan_y=jan_y,
-                         xlo=[xlo],
-                         xhi=[xhi],
-                         ylo=[ylo],
-                         yhi=[yhi],
-                         sampling=[sampling],
-                         axon_lambda=[axon_lambda])
-
-        self.eye = eye
-        self.axon_lambda = axon_lambda
-        self.sampling = sampling
-        self.axon_id = ax_id
-        self.axon_weight = ax_wt
-        self.jan_x = jan_x
-        self.jan_y = jan_y
-        self.range_x = self.gridx.max() - self.gridx.min()
-        self.range_y = self.gridy.max() - self.gridy.min()
-
-    def current2effectivecurrent(self, cs):
+                six.moves.cPickle.dump(grid_dict, open(filename, 'wb'))
+        else:
+            # Assign all elements in the loaded dictionary to this object
+            for key, value in six.iteritems(load_grid_dict):
+                setattr(self, key, value)
+
+    def current2effectivecurrent(self, current_spread):
         """
 
         Converts a current spread map to an 'effective' current spread map, by
         passing the map through a mapping of axon streaks.
 
         Parameters
         ----------
-        cs : array
+        cs: array
             The 2D spread map in retinal space
 
         Returns
         -------
-        ecm : array
-            The effective current spread, a time-series of the same size as the
-            current map, where each pixel is the dot product of the pixel
+        ecm: array
+            The effective current spread, a time - series of the same size as
+            the current map, where each pixel is the dot product of the pixel
             values in ecm along the pixels in the list in axon_map, weighted
             by the weights axon map.
         """
-        ecs = np.zeros(cs.shape)
-        for id in range(0, len(cs.flat)):
-            ecs.flat[id] = np.dot(cs.flat[self.axon_id[id]],
-                                  self.axon_weight[id])
-
-        # normalize so the response under the electrode in the ecs map
-        # is equal to cs
-        scale = np.max(cs) / ecs.flat[np.argmax(cs)]
-        ecs = ecs * scale
-
-        # this normalization is based on unit current on the retina producing
```

Comment:
Do we want to retain some of this comment?

Reply:
On L.296 we say "Normalize so that the max of `ecs` is the same as `current_spread`"

Reply:
Sure. It just seemed like the old comment contained a bit more information than that.